File: app/src/main/java/me/anon/model/StageChange.java
package me.anon.model;

/**
 * // TODO: Add class description
 *
 * @author 7LPdWcaW
 * @documentation // TODO Reference flow doc
 * @project GrowTracker
 */
public class StageChange extends Action
{
	private PlantStage newStage;

	public StageChange()
	{
	}

	public StageChange(PlantStage stage)
	{
		this.setDate(System.currentTimeMillis());
		this.setNewStage(stage);
	}

	public PlantStage getNewStage()
	{
		return newStage;
	}

	public void setNewStage(PlantStage newStage)
	{
		this.newStage = newStage;
	}
}

File: app/src/main/java/me/anon/model/StageChange.kt
package me.anon.model

import com.squareup.moshi.JsonClass
import kotlinx.android.parcel.Parcelize

@Parcelize
@JsonClass(generateAdapter = true)
class StageChange(
	var newStage: PlantStage = PlantStage.PLANTED,

	override var date: Long = System.currentTimeMillis(),
	override var notes: String? = null
) : Action(date, notes)
{
	public var type: String = "StageChange"
}
